data/transactions/logic/eval.go

```diff
@@ -3154,3 +3199,124 @@ func opAppParamsGet(cx *evalContext) {
 	cx.stack[last] = value
 	cx.stack = append(cx.stack, stackValue{Uint: exist})
 }
+
+func authorizedSender(cx *evalContext, addr basics.Address) bool {
+	appAddr, err := cx.getApplicationAddress()
+	return err == nil && bytes.Equal(addr[:], appAddr)
+	// This should also accept accounts that have been rekeyed to the appAddr
+}
+
+func opTxBegin(cx *evalContext) {
+	// Start fresh
+	cx.subtxn = transactions.Transaction{}
+	// Fill in defaults.
+	bytes, err := cx.getApplicationAddress()
+	if err != nil {
+		cx.err = err
+		return
+	}
+	var addr basics.Address
+	copy(addr[:], bytes)
+	cx.subtxn.Header = transactions.Header{
+		Sender:     addr,
+		Fee:        basics.MicroAlgos{Raw: cx.Proto.MinTxnFee},
```

Comment:
The app's account is charged a min fee for using tx_perform.  Since we now allow fee pooling, a better idea is to allow "leftover" fees from the transactions in the group to cover these fees first.

Reply:
I think we would allow `txfield Fee` if we want Fee settable.  My real plan for Fee is:

Add up all the fees in the group, make sure it is at least n*min_fee. (That is standard fee pooling.)
For any amount that it exceeds that minimum, keep track, and use that overage first, rather than charge a fee in these txns.

I don't know if it's worth doing that though.  I guess the nice this is that it means this stuff would be usable without prefunding the app account.

Reply:
Latest commit implements the fee pooling into the app transactions.